serde_codegen/src/attr.rs

```diff
@@ -82,17 +82,19 @@ impl ContainerAttrs {
                     // Parse `#[serde(rename="foo")]`
                     ast::MetaItemKind::NameValue(ref name, ref lit) if name == &"rename" => {
                         let s = try!(get_str_from_lit(cx, name, lit));
-
                         container_attrs.name.serialize_name = Some(s.clone());
                         container_attrs.name.deserialize_name = Some(s);
                     }
 
                     // Parse `#[serde(rename(serialize="foo", deserialize="bar"))]`
                     ast::MetaItemKind::List(ref name, ref meta_items) if name == &"rename" => {
                         let (ser_name, de_name) = try!(get_renames(cx, meta_items));
-
-                        container_attrs.name.serialize_name = ser_name;
-                        container_attrs.name.deserialize_name = de_name;
+                        if ser_name.is_some() {
+                            container_attrs.name.serialize_name = ser_name;
```

Comment:
Good call. Bedtime for me but I filed #355 and I will try to follow up this week.